### PR TITLE
Adds the XML defaults to run an ne30 case for CIME

### PR DESCRIPTION
**Pull request by AaronDonahue**
This commit will make it possible to build an ne30 testcase for
SCREAMv1 and run out-of-the-box.

It does this by adding the hgrid conditional to some of the XML
entries.

Replaces #1587

### CODE REVIEW

File: components/scream/cime_config/namelist_defaults_scream.xml [hunk 1]
```diff
@@ -145,8 +145,9 @@ tweaking their $case/namelist_scream.xml file.
     <!-- Homme dynamics -->
     <homme inherit="atm_proc_base">
       <Grid>Dynamics</Grid>
-      <Vertical__Coordinate__Filename>${DIN_LOC_ROOT}/atm/scream/init/init_ne4np4.nc</Vertical__Coordinate__Filename>
-      <Vertical__Coordinate__Filename COMPSET=".*SCREAM%AQUA.*">${DIN_LOC_ROOT}/atm/scream/init/scream-aquaplanet_init_ne4np4_L72_20211202.nc</Vertical__Coordinate__Filename>
+      <Vertical__Coordinate__Filename hgrid="ne4np4">${DIN_LOC_ROOT}/atm/scream/init/init_ne4np4.nc</Vertical__Coordinate__Filename>
+      <Vertical__Coordinate__Filename hgrid="ne30np4">${DIN_LOC_ROOT}/atm/scream/init/init_ne4np4.nc</Vertical__Coordinate__Filename>
```
Comment (PeterCaldwell) [inline components/scream/cime_config/namelist_defaults_scream.xml:149]: why does the ne30 case read in ne4 IC?

**Comment by AaronDonahue** [comment]
@PeterCaldwell , it should be resolved now.

**Review by PeterCaldwell — APPROVED** [review]

**Comment by AaronDonahue** [comment]
Failed `weaver` but not sure why.  The DIFF's on `mappy` actually make sense given the slight bugfix to SPA.

**Comment by AaronDonahue** [comment]
Okay this time around it passed `weaver` and only failed `mappy` with DIFFs.  As expected.  I'm going to manually merge.

**Pull request merged**